core/src/main/java/com/graphhopper/routing/util/PriorityCode.java

```diff
@@ -53,4 +53,12 @@ public static double getFactor(int value) {
     public static double getValue(int value) {
         return getFactor(value);
     }
+
+    public PriorityCode worse() {
```

Comment:
By using a NavigableSet like a TreeSet, this would be more readable and avoid allocating an array on each call:

```suggestion
    public PriorityCode worse() {
        VALUES.floor(this);
    }
```

Reply:
Arrays are mutable. Therefore any call to `values()` has to create a new array.

https://stackoverflow.com/a/32799324